### ISSUE TITLE
Warning with preprocessor

### ISSUE THREAD

**Issue by atsju** [doxygen/doxygen#7104]
Doxygen version 1.8.15
Following warning is generated with specific preprocessor lines:
> test.h:3: warning: preprocessing issue while doing constant expression evaluation: syntax error

Source code to replicate the issue:
```
#define FAMILY_XX4RFR2_USED	    (defined(__AVR_ATmega644RFR2__)||defined(__AVR_ATmega256RFR2__)||defined(__AVR_ATmega2564RFR2__))

#if !FAMILY_XX4RFR2_USED
    //something
#endif
```
Of course none of `__AVR_ATmega `are defined

This issue may be related to #6943

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7106

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
